File: server/src/main/java/pl/ordovita/surveys/domain/model/userResponse/TextAnswer.java
package pl.ordovita.surveys.domain.model.userResponse;

import pl.ordovita.surveys.domain.exception.TextAnswerException;

public record TextAnswer(String value) {

    public TextAnswer{
        validate(value);
    }


    private void validate(String value) {

        if(value == null || value.isEmpty()) throw new TextAnswerException("Value cannot be null or empty");
        if(value.length() < 3) throw new TextAnswerException("Value must be at least 3 characters");
        if (value.length() > 100) throw new TextAnswerException("Value must be at most 100 characters");
    }
}

File: DotNetServer/Ordovita.Domain/Surveys/UserResponse/TextAnswer.cs
namespace Ordovita.Domain.Surveys.UserResponse;

public record TextAnswer(string Value)
{
    public static TextAnswer New(string value)
    {
        if (string.IsNullOrWhiteSpace(value))
            throw new ArgumentException("Value cannot be null or whitespace.", nameof(value));
        return new TextAnswer(value);
    }

    public static TextAnswer From(string value)
    {
        if (string.IsNullOrWhiteSpace(value))
            throw new ArgumentException("TextAnswer cannot be empty.", nameof(value));
        if (value.Length < 2)
            throw new ArgumentException("TextAnswer must be at least 2 characters.", nameof(value));
        if (value.Length > 100)
            throw new ArgumentException("TextAnswer must be at most 100 characters.", nameof(value));
        return new TextAnswer(value);
    }
}